# Copyright 2025 Reazon Holdings, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# beta 0.25
FollowerNode:
  ros__parameters:
    Dn: [0.7222,   0.7,     0.7,     0.7,     0.12,    0.11,   0.15,   0.050]
    Jn: [0.55,     0.350,   0.10,    0.024,   0.020,   0.013,  0.008,  0.014]
    gn: [23.0,     23.0,    20.0,    29.0,    15.0,    15.0,   15.00,   0.0 ]
    Kp: [1800.000, 2100.000, 5200.000, 5300.000, 15000.000, 500.000, 300.000, 350.000]
    Kd: [22.000,   24.000,  67.000,  55.000,  200.000,  20.000, 20.000, 20.000]
    Kf: [3.500,    3.100,   2.000,   2.90,   0.800,   1.00,  1.00,  0.00]
    Fc: [0.336,    0.336,   0.450,   0.146,   0.070,   0.093,  0.212,  0.212]
    k:  [28.417,  28.417,  29.065,  130.038, 151.771, 242.287,7.888,  7.0  ]
    Fv: [0.063,    0.063,    0.604,   0.813,   0.029,   0.072,  0.084,  0.084]
    Fo: [0.088,    0.088,    0.008,  -0.058,   0.005,   0.009, -0.059, -0.05 ]

# beta 1.0
# FollowerNode:
#   ros__parameters:
#     Dn: [0.7222,   0.7,     0.7,     0.7,     0.12,    0.11,   0.15,   0.050]
#     Jn: [0.55,     0.350,   0.10,    0.024,   0.020,   0.013,  0.008,  0.014]
#     gn: [19.0,     23.0,    19.0,    26.0,    15.0,    15.0,   15.00,   10.0 ]
#     Kp: [3000.000, 2100.000, 5200.000, 5300.000, 1000.000, 500.000, 300.000, 350.000]
#     Kd: [35.000,   24.000,  67.000,  55.000,  60.000,  20.000, 20.000, 20.000]
#     Kf: [2.900,    2.900,   1.800,   2.00,   0.800,   0.800,  1.00,  0.000]
#     Fc: [0.336,    0.336,   0.450,   0.146,   0.070,   0.093,  0.212,  0.212]
#     k:  [28.417,  28.417,  29.065,  130.038, 151.771, 242.287,7.888,  7.0  ]
#     Fv: [0.063,    0.063,    0.604,   0.813,   0.029,   0.072,  0.084,  0.084]
#     Fo: [0.088,    0.088,    0.008,  -0.058,   0.005,   0.009, -0.059, -0.05 ]
